ecllibrary/teststd/uni/TestExcludeNthWord.ecl

```diff
@@ -0,0 +1,65 @@
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+ ############################################################################## */
+ 
+ IMPORT Std.Uni;
+ 
+ EXPORT TestExcludeNthWord := MODULE
+ 
+   EXPORT TestConst := MODULE
+     //Check action on a string with no entries.
+     EXPORT Test01 := ASSERT(Uni.ExcludeNthWord('',0)+'!' = '!', CONST);
+     EXPORT Test02 := ASSERT(Uni.ExcludeNthWord('',1)+'!' = '!', CONST);
+     EXPORT Test03 := ASSERT(Uni.ExcludeNthWord('',-1)+'!' = '!', CONST);
+     EXPORT Test04 := ASSERT(Uni.ExcludeNthWord('             ',0)+'!' = '!', CONST);
+     EXPORT Test05 := ASSERT(Uni.ExcludeNthWord('             ',1)+'!' = '!', CONST);
+     EXPORT Test06 := ASSERT(Uni.ExcludeNthWord('             ',-1)+'!' = '!', CONST);
+     //Check action on a string containing a single word - with various whitespace
+     EXPORT Test07 := ASSERT(Uni.ExcludeNthWord('x',0)+'!' = 'x!');
+     EXPORT Test08 := ASSERT(Uni.ExcludeNthWord('x',1)+'!' = '!');
+     EXPORT Test09 := ASSERT(Uni.ExcludeNthWord('x',2)+'!' = 'x!');
+     EXPORT Test10 := ASSERT(Uni.ExcludeNthWord('x',3)+'!' = 'x!');
+     EXPORT Test11 := ASSERT(Uni.ExcludeNthWord(' x',1)+'!' = '!');
+     EXPORT Test12 := ASSERT(Uni.ExcludeNthWord('x ',1)+'!' = '!');
+     EXPORT Test13 := ASSERT(Uni.ExcludeNthWord(' x',2)+'!' = ' x!');
+     EXPORT Test14 := ASSERT(Uni.ExcludeNthWord(' x ',1)+'!' = '!');
+     //Check action on a string containg multiple words - with various whitespace combinations.
+     EXPORT Test15 := ASSERT(Uni.ExcludeNthWord(' abc def ', 1)+'!' = 'def !');
+     EXPORT Test16 := ASSERT(Uni.ExcludeNthWord(' abc def ', 2)+'!' = ' abc !');
+     EXPORT Test17 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',0)+'!' = '  a b c   def    !');
+     EXPORT Test18 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',1)+'!' = 'b c   def    !');
+     EXPORT Test19 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',2)+'!' = '  a c   def    !');
+     EXPORT Test20 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',3)+'!' = '  a b def    !');
+     EXPORT Test21 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',4)+'!' = '  a b c   !');
+     EXPORT Test22 := ASSERT(Uni.ExcludeNthWord('  a b c   def    ',5)+'!' = '  a b c   def    !');
+     EXPORT Test23 := ASSERT(Uni.ExcludeNthWord(' ,,,, ',1)+'!' = '!');
+     //Test other space characters (< 0x20)
+     EXPORT Test24 := ASSERT(Uni.ExcludeNthWord('  a b\nc \t  def    ',2)+'!' = '  a c \t  def    !');
+     EXPORT Test25 := ASSERT(Uni.ExcludeNthWord('  a b\nc \t  def    ',3)+'!' = '  a b\ndef    !');
+     //Test latin diacritical marks
+     EXPORT Test26 := ASSERT(Uni.ExcludeNthWord(U'À à',2)+U'!' = U'À !');
+     EXPORT Test27 := ASSERT(Uni.ExcludeNthWord(U'ä̰́ Ä̰́',2)+U'!' = U'ä̰́ !');
+     //Test Chinese. "I am a computer" --> "I am"
```

Comment:
Also, it would be worth including tests that include numbers - both as separate words and combined e.g., "abc 123 abc123 123abc"

- Check that words extracted at all word positions are correct/consistent (I suspect this may not be the case).